### PR TITLE
Calculate and print first n Fibonacci numbers and related tests

### PR DESCRIPTION
**Pull request by eobjects**
Fixes #24 

#### Changes proposed in this PR:

- Calculate and print Fibonacci numbers for a given number
- Unit Test cases including negative scenarios

### CODE REVIEW

File: 01_fib/main.go [hunk 1]
```diff
@@ -4,16 +4,33 @@ import (
...
-	a, b := 1, 1
-	for i := 0; i < n; i++ {
-		fmt.Fprintln(out, a)
-		a, b = b, a+b
+	//ideally this can be printed in main but lab needs to be printed inside the func
+	printResult(calculateFib(n))
+}
+
+func printResult(arrayToPrint []int) {
+	resultsAsStr := []string{}
```
Comment (anzdaddy) [inline 01_fib/main.go]: You should output directly to outWriter using `fmt.Sprintf()`.

File: 01_fib/main.go [hunk 2]
```diff
@@ -4,16 +4,33 @@ import (
...
 	"os"
+	"strconv"
+	"strings"
 )
 
-var out io.Writer = os.Stdout
+var outWriter io.Writer = os.Stdout
+
+func calculateFib(n int) []int {
+	f0, f1, result := 1, 1, []int{}
```
Comment (anzdaddy) [inline 01_fib/main.go]: `make([]int, 0, n)`

File: 01_fib/main_test.go [hunk 3]
```diff
@@ -2,69 +2,76 @@ package main
...
+	outWriter = &buf
 
 	// When
-	fib(0)
+	fib(-1)
 
-	//Then
-	r.Equalf("", buf.String(), "Unexpected output in main()")
+	// Then
+	expected := `
```
Comment (anzdaddy) [inline 01_fib/main_test.go]: Prefer no output for empty slice.

File: 01_fib/main_test.go [hunk 4]
```diff
@@ -2,69 +2,76 @@ package main
...
+	outWriter = &buf
 
 	// When
-	fib(-1)
+	fib(0)
 
-	//Then
-	r.Equalf("", buf.String(), "Unexpected output in main()")
+	// Then
+	expected := `
```
Comment (anzdaddy) [inline 01_fib/main_test.go]: Prefer no output for empty slice.

**Review by anzdaddy — DISMISSED** [review]

File: 01_fib/main.go (hunk 1 above)
Comment (eobjects) [reply 01_fib/main.go]: @anzdaddy The output is expected to have new lines, I am not sure if an int slice can be formatted to be printed with new lines

File: 01_fib/main.go [hunk 5]
```diff
@@ -6,13 +6,32 @@ import (
 	"os"
 )
 
-var out io.Writer = os.Stdout
+var outWriter io.Writer = os.Stdout
+
+func calculateFib(n int) []int {
+
```
Comment (juliaogris) [inline 01_fib/main.go]: Remove blank lines at beginning/end of block.

**Review by juliaogris — COMMENTED** [review]
One minor code update and please move your sources to `01_fib/eobjects`

File: 01_fib/srini/main.go [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+	"io"
+	"os"
+)
+
+var outWriter io.Writer = os.Stdout
+
+func calculateFib(n int) []int {
+	//need to check for edge case to avoid array out of range
+	if n < 0 {
+		n = 0
```
Comment (juliaogris) [inline 01_fib/srini/main.go]: or you just abort here and return `[]int{}` for `n <= 0`

**Review by juliaogris — DISMISSED** [review]
LGTM

File: 01_fib/srini/main.go (hunk 6 above)
Comment (eobjects) [reply 01_fib/srini/main.go]: fixed this

**Review by juliaogris — APPROVED** [review]
LGTM

**Pull request merged**